src/data/finset/noncomm_prod.lean

```diff
@@ -226,4 +226,20 @@ begin
   { simp [ha, IH] }
 end
 
+@[to_additive] lemma noncomm_prod_union_of_disjoint [decidable_eq α] {s t : finset α}
+  (h : disjoint s t) (f : α → β)
+  (comm : ∀ (x ∈ s ∪ t) (y ∈ s ∪ t), commute (f x) (f y))
+  (scomm : ∀ (x ∈ s) (y ∈ s), commute (f x) (f y) :=
+    λ _ hx _ hy, comm _ (mem_union_left _ hx) _ (mem_union_left _ hy))
+  (tcomm : ∀ (x ∈ t) (y ∈ t), commute (f x) (f y) :=
+    λ _ hx _ hy, comm _ (mem_union_right _ hx) _ (mem_union_right _ hy)) :
+  noncomm_prod (s ∪ t) f comm = noncomm_prod s f scomm * noncomm_prod t f tcomm :=
```

Comment:
Isn't this true under the weaker assumption that all the elements in `s` commute with all the elements in `t`?